modules/org/infinispan/runtime/added/gossiprouter/bin/launch.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/sh
+# ===================================================================================
+# Entry point for the image which initiates the JGroups Gossip Router
+# ===================================================================================
+
+set -e
+
+JGROUPS_JAR=`find ${ISPN_HOME} -type f -name "jgroups-*.jar" | head -n 1`
+GOSSIP_CLASS="org.jgroups.stack.GossipRouter"
+
+if [ ! -f "${JGROUPS_JAR}" ]; then
+    # TODO create native binary of GossipRouter?
```

Comment:
Ideally this should be a native binary that is contained within it's own minimal image based upon [ub8-micro](https://catalog.redhat.com/software/containers/ubi8-micro/601a84aadd19c7786c47c8ea?container-tabs=overview). Upstream the JGroups project could use a GH action to create the binary and package it in a container. Downstream might be trickier though, WDYT?

Reply:
AFAIK, we don't support native images downstream; I think using the JGroups jar inside our image is the best way:
* It uses the productized JGroups inside our image
* The infinispan server image is built for multiple platforms

Using an image with the Gossip Router (there is an old one here: https://registry.hub.docker.com/r/jboss/jgroups-gossip) it would require changing the operator code for upstream and downstream
* upstream would use the gossip router image
* downstream the infinispan server image

is it worth the trouble? 

IMO, I think Tristan's idea will be simpler: https://infinispan.zulipchat.com/#narrow/stream/118645-infinispan/topic/meeting-20210726/near/247210074

Reply:
> IMO, I think Tristan's idea will be simpler: https://infinispan.zulipchat.com/#narrow/stream/118645-infinispan/topic/meeting-20210726/near/247210074

Isn't that the same as mine? Or do you just mean a native binary stored within our existing image?



> is it worth the trouble?

Probably not :smiley:

If we can't have native binaries downstream then I'm happy to go with this approach for now and we can refactor later as required.

Reply:
I mean storing the GossipRouter binary in our infinispan-server native image.